backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/VdsManager.java

```diff
@@ -1247,10 +1260,37 @@ public void vmsMonitoringInitFinished() {
         if (!isInitialized()) {
             log.info("VMs initialization finished for Host: '{}:{}'", cachedVds.getName(), cachedVds.getId());
             resourceManager.handleVmsFinishedInitOnVds(cachedVds.getId());
+            updateCpuPinning(vmDao.getMonitoredVmsRunningByVds(cachedVds.getId()));
             setInitialized(true);
         }
     }
 
+    private void updateCpuPinning(List<VM> vms) {
+        for (VM vm : vms) {
+            String pinning;
+            switch (vm.getCpuPinningPolicy()) {
+                case MANUAL:
+                    pinning = vm.getCpuPinning();
+                    break;
+                case RESIZE_AND_PIN_NUMA:
+                case DEDICATED:
+                    pinning = vm.getCurrentCpuPinning();
+                    break;
+                default:
+                    return;
+            }
```

Comment:
how about extracting that switch-case to a different method? it seems like a reusable piece of code and it will improve this method..